ragmac_xdem/dem_postprocessing.py

```diff
@@ -771,57 +772,81 @@ def fname_func(pair_id):
             ds = ds.drop(['spatial_ref']) 
             ds.to_zarr(zarr_stack_tmp_fn)
+            print('Zarr file info')
+            source_group = zarr.open(zarr_stack_tmp_fn)
+            source_array = source_group['band1']
+            print(source_group.tree())
+            print(source_array.info)
+            del source_group
+            del source_array
+            
+            print('Removing temporary nc files')
+            for f in Path(dems_list[0]).parents[0].glob('*.nc'):
+                f.unlink(missing_ok=True)
+            
+            print('Creating final zarr stack')
+            print(str(zarr_stack_fn))
             ds = xr.open_dataset(zarr_stack_tmp_fn,
                                  chunks={'time': t, 'y': y, 'x':x},engine='zarr')
             ds['band1'].encoding = {'chunks': (t, y, x)}
             ds.to_zarr(zarr_stack_fn)
+            print('Zarr file info')
+            source_group = zarr.open(zarr_stack_fn)
+            source_array = source_group['band1']
+            print(source_group.tree())
+            print(source_array.info)
+            del source_group
+            del source_array
                 
-            print('removing nc files')
-            for f in Path(dems_list[0]).parents[0].glob('*.nc'):
-                f.unlink(missing_ok=True)
+            print('Removing temporary zarr stack')
+            shutil.rmtree(zarr_stack_tmp_fn, ignore_errors=True)
             
+            print('\nComputing linear regression')
+            print('Check dask dashboard link printed above to monitor progress.')
             ds = xr.open_dataset(zarr_stack_fn,
                                  chunks={'time': t, 'y': y, 'x':x},engine='zarr')
             count_thresh = 5
-            print('\nExcluding pixels with count <',count_thresh)
-            
-            time_delta_min = None
-#             time_delta_min = 5
-#             print('Excluding pixels with max time delta <',time_delta_min, 'days')
-            
-            print('\nCheck dask dashboard link printed above to monitor linear regression progress.')
+            print('Excluding pixels with count <',count_thresh)
+            # if not using matplotlib.dates.date2num to convert time stamps, check delta is computed
+            # and reported correctly
+            min_date = np.percentile(ds.time, 2)
+            max_date = np.percentile(ds.time, 98)
+            time_delta_max = int((max_date - min_date).astype('timedelta64[D]') / np.timedelta64(1, 'D'))
+            time_delta_min = int(time_delta_max * 0.1)
+            print("Min 2 percentile date:",np.datetime_as_string(min_date, unit='D'))
+            print("Max 98 percentile date:",np.datetime_as_string(max_date, unit='D'))
+            print("Time delta between dates:",time_delta_max, 'days')
+            print('Excluding pixels with max time delta <',time_delta_min, 'days')
             results = temporal.dask_apply_linreg(ds['band1'],
```

Comment:
See lines above here for precomputing time delta threshold prior to linear regression.